Add supporters page

choldgraf opened the pull request:
This updates our "supporters" pages so that they are a bit more discoverable and cleanly separated. Here are the major changes:

- Organizations that employ team members are now listed on the "our team" page, to differentiate from cloud/financial support
- We now have a dedicated page to list supporters of cloud/financial
- We store this information in YAML files, so it is easier to programmatically read and can be the source of truth
- The Binder federation page now focuses on the domains and logos, rather than "funded by etc" since we have the supporters page for those thanks now.
- I updates some of our `conf.py` scripts to be easier to read and maintain

manics commented:
RTD failure:
```

Running Sphinx v4.5.0
repo2docker page does not exist, downloading from repo2docker docs. config_files.rst
repo2docker page does not exist, downloading from repo2docker docs. howto/languages.rst
repo2docker page does not exist, downloading from repo2docker docs. howto/user_interface.rst
repo2docker page exists, skipping download. Delete it to re-download. howto/lab_workspaces.rst 
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/mybinder/checkouts/262/doc/_data/scripts/gen_federation_md.py", line 4, in <module>
    import pandas as pd
ModuleNotFoundError: No module named 'pandas'
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/mybinder/checkouts/262/doc/_data/scripts/gen_support_md.py", line 4, in <module>
    import pandas as pd
ModuleNotFoundError: No module named 'pandas'
```

manics commented on doc/_data/scripts/gen_support_md.py:
```diff
@@ -0,0 +1,56 @@
+  # Generate markdown entries for each member
+  supporters = pd.DataFrame(supporters)
+  output = ""
+  for ix, item in supporters.iterrows():
```
Can this be replace by just iterating through `supporters_yaml` without pandas?
```py
for item in supporters_yaml:
    ...
```

manics commented on doc/_data/scripts/gen_federation_md.py:
```diff
@@ -0,0 +1,56 @@
+
+# Generate markdown entries for each federation member
+entries = []
+for ix, binderhub in binderhubs.iterrows():
```
As with supporters I think this can be simplified to avoid pandas
```py
binderhubs_yml = safe_load(path_data.read_text())
for binderhub in binderhubs_yml:
   ...
```

manics commented on doc/about/support.md:
```diff
@@ -54,23 +69,21 @@ In addition to developing and maintaining the many open source tools built by th
-- From 2019 through 2021, `gke.mybinder.org` was supported by cloud credits from [Google Cloud](https://cloud.google.com/).
-- In 2017 the Binder Project received a 1-year grant from the
-`Moore Foundation <https://figshare.com/s/e9d0ad7bdc4e405cccfa>`_.
+Most of the code used by the Binder Project is not hosted in this repository. If you'd like to contribute to the code, check out [the JupyterHub Team Compass guide to contributing](https://jupyterhub-team-compass.readthedocs.io/en/latest/team/skills.html).
```
The skills link is included in the above section, so might be good to link to relevant repositories, or perhaps indicate that https://github.com/jupyterhub/binderhub is the starting point?

choldgraf commented:
Ah good call - I've removed the pandas code from those scripts, and have also added a cross-link to the "our projects" page and links to the github repositories.

manics commented on doc/_data/support/supporters.yml:
```diff
@@ -0,0 +1,49 @@
+    logo: https://github.com/2i2c-org/2i2c-org.github.io/blob/main/static/media/logo.png?raw=true
+    url: https://2i2c.org
+
+  - name: University of Dundee
+    logo: https://upload.wikimedia.org/wikipedia/en/c/cb/University_of_Dundee_shield.png
+    url: https://www.dundee.ac.uk/
+
```
```suggestion
```
It's mostly unofficial for me :smile:

manics reviewed (commented):
Thanks, changes look good to me!

choldgraf commented:
Thanks @manics for the review! I will merge this one in so that we can keep these docs up to date, and will iterate in subsequent PRs if there are more things to change!
